docs(rust): Document strict trace continuation options (#18872)
getsentry/sentry-docs: docs/platforms/rust/common/configuration/options.mdx

</SdkOption>

<SdkOption name="org_id" type='OrganizationId'>

An optional override for the SDK's organization ID.

By default, the organization ID is parsed from the DSN. This only works for Sentry SaaS ingest hosts (`o123.ingest.sentry.io`, including regional hosts such as `o123.ingest.us.sentry.io`). For self-hosted Sentry and local Relay setups, it cannot be derived from the DSN, so set it explicitly with this option.

The organization ID is used for trace continuation, including <PlatformLink to="/configuration/options#strict_trace_continuation">strict trace continuation</PlatformLink>.

</SdkOption>

<SdkOption name="strict_trace_continuation" type='bool' defaultValue='false'>

When enabled, the SDK only continues an incoming trace if it contains an organization ID that matches the current Sentry client's.

Third-party services instrumented with Sentry by another organization can send traces to your application through incoming `sentry-trace` and `baggage` headers. To keep those separate from your own traces, the SDK compares the incoming `sentry-org_id` (from the `baggage` header) with the client's organization ID.

> Whenever both organization IDs are known and differ, the SDK starts a new trace instead of continuing the incoming one, regardless of this option. Enabling this option extends that to traces carrying no organization ID: they are never continued. When disabled (the default), such traces are.

Reviewer: > When disabled (the default), such traces are.

l: Is the sentence finished prematurely here or am I missing something?

Author: It's finished but it only makes sense if you read the previous sentence. Perhaps it could be made more clear? 

It's basically saying that if there's no incoming org ID, we continue traces only when strict trace continuation is disabled.

Reviewer 2: ```suggestion
Whenever both organization IDs are known and differ, the SDK starts a new trace instead of continuing the incoming one, regardless of this option. Enabling this option also prevents the SDK from continuing incoming traces that have no organization ID. When disabled (the default), the SDK continues traces with no organization ID.
```
